sssd_test_framework/roles/ad.py

```diff
@@ -2331,3 +2386,393 @@ def print(self) -> str:
 
 
 ADNetgroupMember: TypeAlias = LDAPNetgroupMember[ADUser, ADNetgroup]
+
+
+class ADCertificateAuthority:
+    """
+    Provides helper methods for Active Directory Certificate Authority operations.
+
+    This class allows requesting, revoking, placing/removing certificate holds,
+    and retrieving certificate information via certreq and certutil commands.
+
+    .. code-block:: python
+       :caption: Requesting and revoke a certificate.
+
+       cert, key, csr = ad.ca.request(
+           template="SmartcardLogon",
+           subject="CN=testuser",
+           enrollment_agent_cert_hash="abc123...",
+           requester_name="Administrator"
+       )
+       ad.ca.revoke(cert, reason="key_compromise")
+    """
+
+    def __init__(self, host: ADHost) -> None:
+        """
+        Initialize the AD Certificate Authority helper.
+
+        :param host: Remote AD host.
+        :type host: ADHost
+        """
+        self.host = host
+        self.cli = host.cli
```

Comment:
We need to clean up the folder on teardown. The easiest thing to do, I think would be to rename the directory to something unique c:\pki then in hosts\ad.py, delete the directory in ```def restore```